### ISSUE TITLE
`pkg_deps_tree()`: Circular dependencies with Remotes cause error

### ISSUE THREAD

**Issue by wch** [r-lib/pak#254]
We have a "soft" circular dependency, where shiny suggests shinytest, and shinytest imports shiny. This causes an error in our CI workflow:
https://github.com/rstudio/shiny/runs/1601512829?check_suite_focus=true#step:6:32

Here's the error when I run locally in the Shiny repo:

```
> pak::pkg_deps_tree("local::.", dependencies = TRUE)
Error: Cannot install packages:                                
* local::.: Can't install dependency rstudio/shinytest
* rstudio/shinytest: Can't install dependency rstudio/shiny
* rstudio/shiny: Conflicts with local::.
```

**Comment by gaborcsardi** [comment]
Circular dependencies are fine, but you told pak to install shiny from that local directory, and shinytest needs shiny from `rstudio/shiny` at GitHub, and pak has no way to know that these two are the same, hence the conflict.

If you remove the `Remotes` field from shinytest, then it will work.

I will have to think what to do to resolve the actual issue.

**Comment by wch** [comment]
shinytest actually requires the dev version of shiny, though, so we can't remove it from the Remotes.

We're just copying the pak commands from here:
https://github.com/r-lib/actions/blob/7216d30a5426859594de6d571f6a3cd065f6d840/examples/check-pak.yaml#L58

If there's some other set of commands that would work better, I'd be happy to switch to it.

**Comment by gaborcsardi** [comment]
> shinytest actually requires the dev version of shiny, though, so we can't remove it from the Remotes.

pak will install the shiny from the local dir, so it will have the dev version of Shiny, at least on GHA.

**Comment by wch** [comment]
Right, that would fix the GHA errors, but other shinytest users will still need shiny listed in Remotes.

**Comment by gaborcsardi** [comment]
Sure, it is only a workaround for GHA. 

I would think that not too many people use dev shinytest, so it might be OK to do this. Especially if shinytest specifies a shiny version, because then they will learn that dev shiny is needed.
